android/src/com/mapswithme/maps/location/LocationHelper.java

```diff
@@ -151,7 +151,6 @@ public void onMyPositionModeChanged(int newMode)
   private final LocationState.LocationPendingTimeoutListener mLocationPendingTimeoutListener = () -> {
     if (mActive)
     {
-      stop();
```

Comment:
1. What happens if a location was found when the dialog was already visible on the screen, and the user pressed (a) Stop or (b) Continue? The dialog should _disappear_ if a location was found when the dialog is visible. Ideally, it would be great to display some kind of a progress bar in that dialog, showing that the GPS satellites search is still active.
2. What about disabling the screen/device lock while the location search is active? When it's enabled, most users won't find their position and we get 1-star complaints in stores.
3. Did you properly test it? How it behaves now? Is behavior better than before? What happens if OM goes into the background when this dialog is visible, and then is open (a) normally or (b) via API?

Reply:
1. The dialog doesn't disappear automatically.
2. Location is disabled onPause().
3. I monitored log messages. The location provider is not stopped anymore when dialog appears.

Reply:
1. Please fix it.
2. My question was about the screen lock, not the location provider. We also discussed it in Telegram.
3. Again, the question was about the behavior when the dialog is visible, and the app goes into the background and is restored normally or via an API call.

Reply:
The alternative solution consists of these 3 changes:
1. Remove dialog completely.
2. Search GPS indefinitely.
3. Allow users to stop GPS search and then start it again only when necessary.

I can describe it in a separate issue.

Reply:
1. Fixed. 
2. Screen lock is OK, we have it in LocationHelper itself. I tested and it works.
3. The dialog is not restored.